### ISSUE TITLE
Helm 2.15 doesn't wait for hook job to complete and ignores failure

### ISSUE THREAD

**Issue by floretan** [helm/helm#6767]
We have been using a hook for our helm chart to perform some key tasks when installing / upgrading a release. Up to version 2.14, calling `helm upgrade --install ...` without the `--wait` flag the behavior was the following:

- Wait for the job to complete
- If the job fails, return an exit code indicating an error

Now with version 2.15.1, the same command gives the following behavior:

- Return right away
- The exit code is a success, even if the job failed.

This is particularly relevant when running from a CI tool, as broken releases are wrongly passing. 

Output of `helm version`:
```
Client: &version.Version{SemVer:"v2.15.1", GitCommit:"cf1de4f8ba70eded310918a8af3a96bfe8e7683b", GitTreeState:"clean"}
Server: &version.Version{SemVer:"v2.15.1", GitCommit:"cf1de4f8ba70eded310918a8af3a96bfe8e7683b", GitTreeState:"clean"}
```
Output of `kubectl version`:
```
Client Version: version.Info{Major:"1", Minor:"16", GitVersion:"v1.16.2", GitCommit:"c97fe5036ef3df2967d086711e6c0c405941e14b", GitTreeState:"clean", BuildDate:"2019-10-15T19:18:23Z", GoVersion:"go1.12.10", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"14+", GitVersion:"v1.14.6-gke.1", GitCommit:"61c30f98599ad5309185df308962054d9670bafa", GitTreeState:"clean", BuildDate:"2019-08-28T11:06:42Z", GoVersion:"go1.12.9b4", Compiler:"gc", Platform:"linux/amd64"}
```
Cloud Provider/Platform (AKS, GKE, Minikube etc.): GKE

**Comment by mfilotto** [comment]
Same for us

**Comment by tedohayer** [comment]
+1, helm did not wait for successful database migration before rolling out a new replicaset

**Comment by thomastaylor312** [comment]
So many of these reported hook issues may all be resolved by a recently merged fix #6897. Can anyone give the `dev-2` branch a try and see if it fixes it for them?

**Comment by bacongobbler** [comment]
While I don't have a chart which I can use to verify the behaviour locally, I'm fairly certain this has been fixed in #6897. That patch fixes an issue with the kube client unable to wait for Jobs to run to completion as seen in #6894. Kubernetes 1.16 introduced a breaking change to the legacy schema so we had to update the code to reflect that change.

I'm going to close this as tentatively resolved via #6897, which will be in Helm 2.16.1. Please feel free to report your findings once 2.16.1 has been released with the likely fix mentioned above and we'll re-open if this is still an issue in 2.16.1+. I'm interested to hear your findings!

**Comment by thomastaylor312** [comment]
I re-tested this and found out what was going on. I'll have a fix soon
